pkg/platform/runtime/installer.go

```diff
@@ -130,23 +170,28 @@ func (installer *Installer) Assembler() (Assembler, *failures.Failure) {
 
 	switch artifacts.BuildEngine {
 	case Alternative:
-		return NewAlternativeRuntime(artifacts.Artifacts, installer.runtime.runtimeDir, artifacts.RecipeID)
+		return NewAlternativeInstall(installer.runtime.runtimeDir, artifacts.Artifacts, artifacts.RecipeID)
 	case Camel:
-		return NewCamelRuntime(installer.runtime.commitID, artifacts.Artifacts, installer.runtime.runtimeDir)
+		return NewCamelInstall(installer.runtime.commitID, installer.runtime.runtimeDir, artifacts.Artifacts)
 	case Hybrid:
-		cr, fail := NewCamelRuntime(installer.runtime.commitID, artifacts.Artifacts, installer.runtime.runtimeDir)
+		ci, fail := NewCamelInstall(installer.runtime.commitID, installer.runtime.runtimeDir, artifacts.Artifacts)
 		if fail != nil {
 			return nil, fail
 		}
 
-		return &HybridRuntime{cr}, nil
+		return &HybridInstall{ci}, nil
 	default:
 		return nil, FailRuntimeUnknownEngine.New("installer_err_engine_unknown")
 	}
 }
 
 // InstallArtifacts installs all artifacts provided by a runtime assembler
 func (installer *Installer) InstallArtifacts(runtimeAssembler Assembler) (envGetter EnvGetter, freshInstallation bool, fail *failures.Failure) {
+	err := installer.runtime.StoreBuildEngine(runtimeAssembler.BuildEngine())
+	if err != nil {
+		return nil, false, failures.FailRuntime.Wrap(err, "TODO")
+	}
+
 	if runtimeAssembler.IsInstalled() {
```

Comment:
I think we'll probably want to mark the installation as complete here as well, so projects that were activated before this change get picked up.